xds/src/main/java/io/grpc/xds/sds/FileBasedPluginCredential.java

```diff
@@ -90,6 +91,12 @@ private static DataSource buildDataSourceFromConfigStruct(Struct secretValueStru
   @Override
   public void applyRequestMetadata(
       @Nullable RequestInfo requestInfo, Executor appExecutor, final MetadataApplier applier) {
+    if (!checkSecurityLevel(requestInfo, SecurityLevel.PRIVACY_AND_INTEGRITY)) {
```

Comment:
So this is currently used for SDS which is over an UDS channel. Will this check now break the SDS communication or is UDS considered at `SecurityLevel.PRIVACY_AND_INTEGRITY` ?

Reply:
UDS should be considered secure (i.e., `SecurityLevel.PRIVACY_AND_INTEGRITY`). Do you know if the security level is correctly set for the UDS channel?

Reply:
Don't know. If not, we will have to do that before merging this change to ensure this feature doesn't break. Let me know if you want to discuss